### PR TITLE
[SPARK-48623][CORE] Structured logging migrations [Part 3]

### PR DESCRIPTION
**Pull request by asl3**
### What changes were proposed in this pull request?
This PR makes additional Scala logging migrations to comply with the scala style changes in https://github.com/apache/spark/pull/46947

### Why are the changes needed?
This makes development and PR review of the structured logging migration easier.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Tested by ensuring dev/scalastyle checks pass

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala [hunk 1]
```diff
@@ -63,9 +62,11 @@ case class SessionHolder(userId: String, sessionId: String, session: SparkSessio
   private lazy val planCache: Option[Cache[proto.Relation, LogicalPlan]] = {
     if (SparkEnv.get.conf.get(Connect.CONNECT_SESSION_PLAN_CACHE_SIZE) <= 0) {
       logWarning(
-        s"Session plan cache is disabled due to non-positive cache size." +
-          s" Current value of '${Connect.CONNECT_SESSION_PLAN_CACHE_SIZE.key}' is" +
-          s" ${SparkEnv.get.conf.get(Connect.CONNECT_SESSION_PLAN_CACHE_SIZE)}.")
+        log"Session plan cache is disabled due to non-positive cache size." +
+        log" Current value of '${MDC(LogKeys.CACHE_SIZE_KEY,
```
Comment (gengliangwang) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala:66]:
```suggestion
        log" Current value of '${MDC(LogKeys.CONFIG,
```

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala [hunk 2]
```diff
@@ -248,15 +249,15 @@ case class SessionHolder(userId: String, sessionId: String, session: SparkSessio
   private[connect] def updateAccessTime(): Unit = {
     lastAccessTimeMs = System.currentTimeMillis()
     logInfo(
-      log"Session ${MDC(SESSION_KEY, key)} accessed, " +
-        log"time ${MDC(LAST_ACCESS_TIME, lastAccessTimeMs)} ms.")
+      log"Session ${MDC(LogKeys.SESSION_KEY, key)} accessed, " +
```
Comment (gengliangwang) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala:252]: The key contains user id and session id. Let's follow the style of the other entry:
```
session with userId: ${MDC(LogKeys.USER_ID, userId)} and " +
      log"sessionId: ${MDC(LogKeys.SESSION_ID, sessionId)
```

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala [hunk 3]
```diff
@@ -248,15 +249,15 @@ case class SessionHolder(userId: String, sessionId: String, session: SparkSessio
...
+      log"time ${MDC(LogKeys.LAST_ACCESS_TIME, lastAccessTimeMs)} ms.")
   }
 
   private[connect] def setCustomInactiveTimeoutMs(newInactiveTimeoutMs: Option[Long]): Unit = {
     customInactiveTimeoutMs = newInactiveTimeoutMs
     logInfo(
-      log"Session ${MDC(SESSION_KEY, key)} " +
-        log"inactive timeout set to ${MDC(TIMEOUT, customInactiveTimeoutMs)} ms.")
+      log"Session ${MDC(LogKeys.SESSION_KEY, key)} " +
```
Comment (gengliangwang) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala:259]: ditto

File: core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala [hunk 4]
```diff
@@ -131,19 +131,23 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
...
 
   // This configuration indicates the module to run each Python worker.
   private val workerModule =
     conf.get(PYTHON_WORKER_MODULE).map { value =>
       logInfo(
-        s"Python worker module in PySpark is set to [$value] in '${PYTHON_WORKER_MODULE.key}', " +
-        "using this to start the worker up. Note that this configuration only has an effect when " +
-        s"'${PYTHON_USE_DAEMON.key}' is disabled or the platform is Windows.")
+        log"Python worker module in PySpark is set to ${MDC(LogKeys.VALUE, value)} " +
+        log"in ${MDC(LogKeys.PYTHON_WORKER_MODULE, PYTHON_WORKER_MODULE.key)}, " +
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala:147]:
```suggestion
        log"in ${MDC(LogKeys.CONFIG, PYTHON_WORKER_MODULE.key)}, " +
```

File: core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala [hunk 5]
```diff
@@ -131,19 +131,23 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
...
+        log"using this to start the worker up. Note that this configuration only has " +
+        log"an effect when ${MDC(LogKeys.PYTHON_USE_DAEMON, PYTHON_USE_DAEMON.key)} " +
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala:149]:
```suggestion
        log"an effect when ${MDC(LogKeys.CONFIG2, PYTHON_USE_DAEMON.key)} " +
```

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 6]
```diff
@@ -136,13 +136,13 @@ private[spark] class SparkSubmit extends Logging {
 
   /** Print version information to the log. */
   private def printVersion(): Unit = {
-    logInfo("""Welcome to
+    logInfo(log"""Welcome to
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala:139]: let's ignore this file

File: core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala [hunk 7]
```diff
@@ -998,11 +998,13 @@ private[spark] class DAGScheduler(
     ThreadUtils.awaitReady(waiter.completionFuture, Duration.Inf)
     waiter.completionFuture.value.get match {
       case scala.util.Success(_) =>
-        logInfo("Job %d finished: %s, took %f s".format
-          (waiter.jobId, callSite.shortForm, (System.nanoTime - start) / 1e9))
+        logInfo(log"Job ${MDC(LogKeys.JOB_ID, waiter.jobId)} finished: " +
+          log"${MDC(LogKeys.CALL_SITE_SHORT_FORM, callSite.shortForm)}, took " +
+          log"${MDC(LogKeys.TIME, (System.nanoTime - start) / 1e9)} s")
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala:1003]: let's change it to ms

File: core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala [hunk 8]
```diff
@@ -998,11 +998,13 @@ private[spark] class DAGScheduler(
...
       case scala.util.Failure(exception) =>
-        logInfo("Job %d failed: %s, took %f s".format
-          (waiter.jobId, callSite.shortForm, (System.nanoTime - start) / 1e9))
+        logInfo(log"Job ${MDC(LogKeys.JOB_ID, waiter.jobId)} failed: " +
+          log"${MDC(LogKeys.CALL_SITE_SHORT_FORM, callSite.shortForm)}, took " +
+          log"${MDC(LogKeys.TIME, (System.nanoTime - start) / 1e9)} s")
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala:1007]: let's change it to ms

File: core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala [hunk 9]
```diff
@@ -131,19 +131,23 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
   private val daemonModule =
     conf.get(PYTHON_DAEMON_MODULE).map { value =>
       logInfo(
-        s"Python daemon module in PySpark is set to [$value] in '${PYTHON_DAEMON_MODULE.key}', " +
-        "using this to start the daemon up. Note that this configuration only has an effect when " +
-        s"'${PYTHON_USE_DAEMON.key}' is enabled and the platform is not Windows.")
+        log"Python daemon module in PySpark is set to " +
+        log"[${MDC(LogKeys.VALUE, value)}] in '${MDC(LogKeys.PYTHON_DAEMON_MODULE,
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala:135]:
```suggestion
        log"[${MDC(LogKeys.VALUE, value)}] in '${MDC(LogKeys.CONFIG,
```

File: core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala [hunk 10]
```diff
@@ -131,19 +131,23 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
...
+          PYTHON_DAEMON_MODULE.key)}', using this to start the daemon up. Note that this " +
+          log"configuration only has an effect when '${MDC(LogKeys.PYTHON_USE_DAEMON,
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala:137]:
```suggestion
          log"configuration only has an effect when '${MDC(LogKeys.CONFIG2,
```

**Review by gengliangwang — APPROVED** [review]

**Comment by panbingkun** [comment]
It's unbelievable that there are still so many unfinished migrations. 
It seems that we still `missed` some special writing methods that caused `grep` to fail.

LGTM.

**Comment by panbingkun** [comment]
Is this the last part?

**Comment by gengliangwang** [comment]
> Is this the last part?

Hopefully yes

**Comment by panbingkun** [comment]
> > Is this the last part?
> 
> Hopefully yes

If need help, please call me and let's speed up the process. @asl3 @gengliangwang

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**